components/vbox/index.js

```diff
@@ -0,0 +1,89 @@
+import Styles from './styles';
+/**
+ * Launches VBox when running Maxthon6.
+ *
+ * @see http://mxdoc.glitch.me/#/payment?id=specifications
+ *
+ * @param {*} props
+ */
+const VBoxView = props => {
+	const outputs = props.outputs.map(each => ({
+		currency: 'BSV',
+		...each
+	}));
+	const appName = props.appName ? props.appName : 'Twetch Pay';
+	const appLogo = props.appLogo ? props.appLogo : 'https://twetch-app.now.sh/static/favi.png';
+	const productName = props.productName ? props.productName : 'Purchase';
+	let adaptedOutputs = [];
+	const MAX_SECONDS_EXPIRY = 120;
+	for (const output of outputs) {
+		adaptedOutputs.push({
+			address: output.to,
+			value: output.amount * 100000000, // Convert to satoshis
+		})
+	}
+	const launchWallet = async  () =>  {
+		if (window && !(window['VBox'])) {
+			throw new Error('VBox not found');
+		}
+		// Construct Request Data
+		const expireTime = (new Date()).getTime() + (MAX_SECONDS_EXPIRY * 1000);
+		const data = {
+			v: 1,
+			app: {
+				name: appName,
+				logo: appLogo,
+			},
+			id: (Math.floor(Math.random() * 9007199254740990) + 1) + '', // Random ID
+			product: productName,
+			to: adaptedOutputs,
+			broadcast: true,
+			ttl: expireTime,
+			expire: expireTime
+		};
+		if (props.debug) {
+			console.log('VBox Data', data);
+		}
+		// Construct Request with base64 encoded Data
+		let requestEnvelope = {
+			pay_request: {
+				data: Buffer.from(JSON.stringify(data)).toString("base64")
+			}
+		};
+		if (props.debug) {
+			console.log('VBox Request', requestEnvelope);
+		}
+		window['VBox'].Request(requestEnvelope, function(result) {
+			try {
+				if (props.debug) {
+					console.log('VBox Data', data);
+				}
+				let response = Buffer.from(result, 'base64').toString("utf8");
+				response = JSON.parse(response);
+				// If there was an error (non-zero)
+				if (response.code !== 0) {
+					console.log('error', response);
+					return;
+				}
+				if (props.debug) {
+					console.log('VBox Data', data);
+					console.log('VBox Response', response);
+				}
+				props.onPayment({
+					txid: response.txhash,
+					rawtx: null,
```

Comment:
does maxthon not return the rawtx? If not i'll reach out to Jeff Chen.

Aslo, we had a request from Joshua Henslee to return the whole response from wallets as well. Can you add `walletResponse: response` to this `onPayment` callback?

Reply:
<img width="843" alt="Screen Shot 2020-06-14 at 9 17 09 PM" src="https://user-images.githubusercontent.com/8557965/84609315-88aa4e00-ae84-11ea-94e3-5d31dabfadd0.png">
More context from mattermost https://mm.twetch.com/twetch/pl/b5fsx97zejbazbkcot5e6xc88w

Reply:
>does maxthon not return the rawtx? If not i'll reach out to Jeff Chen.

No they do not return it.

Reply:
>Can you add walletResponse: response to this onPayment callback?

Yes, adding...